languages/java/test/ServiceResultTest.java

```diff
@@ -0,0 +1,174 @@
+        assertInstanceOf(Date.class, serviceResultWithSampleMessage.getDate());
+        assertEquals(TestUtils.SAMPLE_SERVICE_RESULT_MESSAGE, serviceResultWithSampleMessage.getMessage());
+    }
+
+    @Test
+    void getId() {
+        ServiceResult serviceResult = ServiceResult.createServiceResult();
+
+        UUID id1 = serviceResult.getId();
+        assertInstanceOf(UUID.class, id1);
+
+        UUID id2 = serviceResult.getId();
+        assertInstanceOf(UUID.class, id2);
+
+        assertSame(id1, id2);
+    }
+
+    @Test
+    void getDate() {
+        ServiceResult serviceResult = ServiceResult.createServiceResult();
+
+        Date date1 = serviceResult.getDate();
+        assertInstanceOf(Date.class, date1);
+
+        Date date2 = serviceResult.getDate();
+        assertInstanceOf(Date.class, date2);
+
+        assertSame(date1, date2);
+    }
+
+    @Test
+    void getMessage() {
+        Map<ServiceResult, String> serviceResults = new HashMap<>();
+
+        serviceResults.put(ServiceResult.createServiceResult(), null);
+        serviceResults.put(ServiceResult.createServiceResult(null), null);
+        serviceResults.put(ServiceResult.createServiceResult(""), "");
+        serviceResults.put(ServiceResult.createServiceResult(TestUtils.SAMPLE_SERVICE_RESULT_MESSAGE), TestUtils.SAMPLE_SERVICE_RESULT_MESSAGE);
+
+        serviceResults.forEach((serviceResult, expectedMessage) -> {
+            String message1 = serviceResult.getMessage();
+            assertEquals(expectedMessage, message1);
+
+            String message2 = serviceResult.getMessage();
+            assertEquals(expectedMessage, message2);
+
+            assertSame(message1, message2);
+        });
+    }
+
+    @Test
+    void testToString() {
+        String expectedStringRegexTemplate = "(?i)^ServiceResult: \\[id: [a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}, date: [a-z]{3} [a-z]{3} \\d{2} \\d{2}:\\d{2}:\\d{2} [a-z]{3} \\d{4,}, message: %s]$";
+
+        List.of(
+                TestUtils.SERVICE_RESULT_WITH_NO_MESSAGE,
+                TestUtils.SERVICE_RESULT_WITH_NULL_MESSAGE,
+                TestUtils.SERVICE_RESULT_WITH_EMPTY_MESSAGE,
+                TestUtils.SERVICE_RESULT_WITH_SAMPLE_MESSAGE
+        ).forEach(serviceResult -> {
+            String serviceResultString = serviceResult.toString();
+            assertInstanceOf(String.class, serviceResultString);
+            assertTrue(serviceResultString.matches(expectedStringRegexTemplate.formatted(serviceResult.getMessage())));
+        });
+    }
+
+    @Test
+    void testEquals() {
+        ServiceResult serviceResult = ServiceResult.createServiceResult();
+        assertInstanceOf(ServiceResult.class, serviceResult);
+
+        ServiceResult serviceResultOther = ServiceResult.createServiceResult();
+        assertInstanceOf(ServiceResult.class, serviceResultOther);
+
+        assertNotEquals(serviceResult, serviceResultOther);
```

Comment:
Even though we created our ServiceResult data class in a way that prevented consumers from easily setting our internally generated property values (`id` and `date`), but maybe we could use reflection here to set values manually and confirm equality via `assertEquals` and `assertNotSame` instead of just relying on a single `assertNotEquals` check.